### PR TITLE
support dropdown on combobox

### PR DESCRIPTION
**Pull request by LawyZheng**
- open the dropdown before scraping so we can get the options in the scraped_page.
- send the option value to LLM, helping it to make the right choice.

### CODE REVIEW

File: skyvern/webeye/scraper/domUtils.js [hunk 1]
```diff
@@ -601,6 +618,21 @@ function buildTreeFromBody() {
     } else if (attrs["role"] && attrs["role"].toLowerCase() === "listbox") {
       // if "role" key is inside attrs, then get all the elements with role "option" and get their text
       selectOptions = getListboxOptions(element);
+    } else if (isComboboxDropdown(element)) {
+      // open combobox dropdown to get options
+      element.click();
```
Comment (wintonzheng) [inline skyvern/webeye/scraper/domUtils.js:623]: very interesting - interacting with the dropdown elements in order to get/parse dropdown elements.

would element ids / unique ids be assigned to the dropdown elements? would line 629 (pressing tab) makes those elements gone. if that happens, then the dropdown element ids we're sending to llm won't be interactable anymore

Comment (wintonzheng) [reply skyvern/webeye/scraper/domUtils.js:623]: makes sense

File: skyvern/webeye/scraper/domUtils.js [hunk 2]
```diff
@@ -601,6 +618,21 @@ function buildTreeFromBody() {
...
+      const listBox = document.getElementById(
```
Comment (wintonzheng) [inline skyvern/webeye/scraper/domUtils.js:624]: is it possible that this listbox will be undefined / null? shall we handle that

**Review by wintonzheng — APPROVED** [review]
logic looks good. let's make sure we cover edge cases. For example, handle all the null cases in this PR so that it won't break dom parsing

File: skyvern/webeye/scraper/domUtils.js (hunk 2 above)
Comment (LawyZheng) [reply skyvern/webeye/scraper/domUtils.js:624]: Yeah, you're right. We should handle the null exception here, tho the possibility to be null is very low.

**Pull request merged**